providers/teams/pages/members/[id].tsx

```diff
@@ -0,0 +1,215 @@
+
+    if (data.users.length > 0) {
+        const userId = data.users[0].id
+        console.log(`userId ${userId}`);
+
+
+        const hasTeamId = await checkHasTeam(userId)
+        if (hasTeamId) {
+            console.log(`${userMail} is already in team ${hasTeamId}`);
+        } else {
+            // add user to team
+            const { data, errors } = await apolloClient.mutate({
+                mutation: insertTeamMember,
+                variables: { objects: { teamId: parseInt(teamId), userId } }
+            })
+            if (errors) {
+                alert(errors[0])
+                throw errors[0]
+            }
+            return 'success'
+        }
+    } else {
+        // user not found
+        return null
+    }
+}
+
+
+const removeMember = async (memberData) => {
+    // remove user from team
+    console.log('removing user')
+    console.log(memberData);
+
+    const { data, errors } = await apolloClient.mutate({
+        mutation: deleteTeamMember,
+        variables: { uid: memberData.user.id, tid: memberData.teamId }
+    })
+    console.log(data)
+    console.log(data.delete_team_members.returning);
+    if (errors) {
+        alert(errors[0])
+        throw errors[0]
+    }
+    return 'success'
+}
+
+
+
+export const TeamEditMembersPage = () => {
+
+
+    const router = useRouter();
+    const [teamId, setTeamId] = useState((typeof router.query.id === 'string') ? router.query.id : router.query.id[0]);
+    const [data, setData] = useState(null)
+    const [teamMembers, setTeamMembers] = useState([]);
+
+    let error;
+    useEffect(() => {
+        console.log("loaded")
+        apolloClient.query({
+            query: getTeamMembers,
+            variables: {
+                teamId
+            }
+        }).then(({ loading, data }) => {
+            if (!loading && data && data.team_members) {
+                setData(data)
+            }
+        }).catch((err) => error = err)
+    }, [])
+
+    useEffect(() => {
+        console.log("2 called")
+        if (data) {
+            setTeamMembers(data.team_members);
+        }
+    }, [data]);
+
+    if (error || !teamMembers) {
+        console.error(error || "Couldn't retrieve data.")
+        return <div>Couldn't retrieve data.</div>
+    }
+
+    async function onSubmit(event: ChangeEvent<HTMLFormElement>) {
+        event.preventDefault()
+        const formElement = document.forms[0]
+        const formData = new FormData(formElement)
+
+        const userMail: string = formData.get('email').toString()
+        const userId = await addMember(teamId, userMail)
+        if (userId) {
+            router.replace(`/teams/members/${teamId}`)
+        }
+    }
+
+    return (
+        <div>
+            <h1>Edit Members Screen</h1>
+            <h3>Team {teamId}</h3>
+            {teamMembers.map((member, i) => (
+                <Member key={i} member={member} teamMembers={teamMembers} setTeamMembers={setTeamMembers} removeMember={removeMember} />
```

Comment:
@shakedlokits if the functions exist in outer scope, should one pass them into props? i would recomend against, what do you think?